bot/build.sh

```diff
@@ -0,0 +1,204 @@
+if [[ ! -z "${CPU_TARGET}" ]]; then
+    EESSI_SOFTWARE_SUBDIR_OVERRIDE=${CPU_TARGET}
+else
+    EESSI_SOFTWARE_SUBDIR_OVERRIDE=$(cfg_get_value "architecture" "software_subdir")
+fi
+export EESSI_SOFTWARE_SUBDIR_OVERRIDE
+
+# get EESSI_OS_TYPE from .architecture.os_type in cfg/job.cfg (default: linux)
+EESSI_OS_TYPE=$(cfg_get_value "architecture" "os_type")
+export EESSI_OS_TYPE=${EESSI_OS_TYPE:-linux}
+
+# TODO
+#   - CODED add handling of EESSI_SOFTWARE_SUBDIR_OVERRIDE to eessi_container.sh
+#     TODO ensure that the bot makes use of that. (currently sets env var
+#     CPU_TARGET & adds --export=ALL,CPU_TARGET=val to sbatch command ... also
+#     add it to cfg/job.cfg - .architecture.software_subdir)
+#   - CODED add handling of http(s)_proxy to eessi_container.sh, in there needs the
+#     CVMFS_HTTP_PROXY added to /etc/cvmfs/default.local (this needs a robust
+#     way to determine the IP address of a proxy)
+#   - bot needs to make repos.cfg and cfg_bundle available to job (likely, by copying
+#     files into './cfg/.' and defining '.repository.repos_cfg_dir' in './cfg/job.cfg')
+
+# prepare options and directories for calling eessi_container.sh
+declare -a BUILD_STEP_ARGS=()
+BUILD_STEP_ARGS+=("--verbose")
+BUILD_STEP_ARGS+=("--access" "rw")
+BUILD_STEP_ARGS+=("--mode" "run")
+BUILD_STEP_ARGS+=("--save" "${PWD}/previous_tmp/build_step")
+BUILD_STEP_ARGS+=("--storage" "${STORAGE}")
+declare -a TARBALL_STEP_ARGS=()
+TARBALL_STEP_ARGS+=("--verbose")
+TARBALL_STEP_ARGS+=("--access" "rw")
+TARBALL_STEP_ARGS+=("--mode" "run")
+TARBALL_STEP_ARGS+=("--save" "${PWD}/previous_tmp/tarball_step")
+CONTAINER_OPT=
+if [[ ! -z ${CONTAINER} ]]; then
+    CONTAINER_OPT="--container ${CONTAINER}"
+    BUILD_STEP_ARGS+=("--container" "${CONTAINER}")
+    TARBALL_STEP_ARGS+=("--container" "${CONTAINER}")
+fi
+HTTP_PROXY_OPT=
+if [[ ! -z ${HTTP_PROXY} ]]; then
+    HTTP_PROXY_OPT="--http-proxy ${HTTP_PROXY}"
+    BUILD_STEP_ARGS+=("--http-proxy" "${HTTP_PROXY}")
+    TARBALL_STEP_ARGS+=("--http-proxy" "${HTTP_PROXY}")
+fi
+HTTPS_PROXY_OPT=
+if [[ ! -z ${HTTPS_PROXY} ]]; then
+    HTTPS_PROXY_OPT="--https-proxy ${HTTPS_PROXY}"
+    BUILD_STEP_ARGS+=("--https-proxy" "${HTTPS_PROXY}")
+    TARBALL_STEP_ARGS+=("--https-proxy" "${HTTPS_PROXY}")
+fi
+REPOSITORY_OPT=
+if [[ ! -z ${REPOSITORY} ]]; then
+    REPOSITORY_OPT="--repository ${REPOSITORY}"
+    BUILD_STEP_ARGS+=("--repository" "${REPOSITORY}")
+    TARBALL_STEP_ARGS+=("--repository" "${REPOSITORY}")
+fi
+GENERIC_OPT=
+if [[ ${EESSI_SOFTWARE_SUBDIR_OVERRIDE} =~ .*/generic$ ]]; then
+    GENERIC_OPT="--generic"
+fi
+
+mkdir -p previous_tmp/{build_step,tarball_step}
+build_outerr=$(mktemp build.outerr.XXXX)
+echo "Executing command to build software:"
+echo "./eessi_container.sh ${BUILD_STEP_ARGS[@]}"
+echo "                     -- ./install_software_layer.sh ${GENERIC_OPT} \"$@\" 2>&1 | tee -a ${build_outerr}"
+# set EESSI_REPOS_CFG_DIR_OVERRIDE to ./cfg
+export EESSI_REPOS_CFG_DIR_OVERRIDE=${PWD}/cfg
```

Comment:
This should be removed, already done above

Reply:
Addressed in https://github.com/EESSI/software-layer/pull/233/commits/d08bd211ed6c23c5b917b01bd01ef75a5cb7b1b9